pkg/template/template.go

```diff
@@ -169,6 +169,10 @@ func (p *Processor) GenerateParameterValues(t *api.Template) error {
 				return fmt.Errorf("template.parameters[%v]: Unable to convert the generated value '%#v' to string", i, value)
 			}
 		}
+		if len(param.Value) == 0 && param.Required {
+			return fmt.Errorf("template.parameters[%v]: parameter is marked required but has no default value and no generator", i)
```

Comment:
Yeah.  At least fix this new one and have a follow up to fix the rest

On Jul 30, 2015, at 8:24 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/template/template.go
<https://github.com/openshift/origin/pull/3963#discussion_r35936756>:

> @@ -169,6 +169,10 @@ func (p *Processor) GenerateParameterValues(t *api.Template) error {
>  				return fmt.Errorf("template.parameters[%v]: Unable to convert the generated value '%#v' to string", i, value)
>  			}
>  		}
> +		if len(param.Value) == 0 && param.Required {
> +			return fmt.Errorf("template.parameters[%v]: parameter is marked required but has no default value and no generator", i)

That's probably true of all the parameter errors if we expect a client to
be able to programmatically identify and present the error associated with
the field

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3963/files#r35936756>.